File: backend/src/main/java/com/debtcollection/model/CaseState.java
package com.debtcollection.model;

import com.fasterxml.jackson.annotation.JsonValue;
import com.fasterxml.jackson.annotation.JsonCreator;

// USER PREFERENCE: Enum per gestire gli stati dei casi di recupero crediti
// CUSTOM IMPLEMENTATION: Added Jackson annotations for proper HTTP parameter serialization
public enum CaseState {
    MESSA_IN_MORA_DA_FARE,
    MESSA_IN_MORA_INVIATA,
    DEPOSITO_RICORSO,
    DECRETO_INGIUNTIVO_DA_NOTIFICARE,
    DECRETO_INGIUNTIVO_NOTIFICATO,
    CONTESTAZIONE_DA_RISCONTRARE,
    PIGNORAMENTO,
    PRECETTO,
    COMPLETATA;

    // CUSTOM IMPLEMENTATION: Jackson serialization for HTTP parameters
    @JsonValue
    public String toValue() {
        return this.name();
    }

    // CUSTOM IMPLEMENTATION: Jackson deserialization for HTTP parameters  
    @JsonCreator
    public static CaseState fromValue(String value) {
        if (value == null) {
            return null;
        }
        try {
            return CaseState.valueOf(value.toUpperCase());
        } catch (IllegalArgumentException e) {
            // Handle case-insensitive matching
            for (CaseState state : CaseState.values()) {
                if (state.name().equalsIgnoreCase(value)) {
                    return state;
                }
            }
            throw new IllegalArgumentException("Invalid CaseState value: " + value);
        }
    }
}

File: frontend/lib/models/case_state.dart
import 'package:json_annotation/json_annotation.dart';

enum CaseState {
  @JsonValue('MESSA_IN_MORA_DA_FARE')
  messaInMoraDaFare,
  @JsonValue('MESSA_IN_MORA_INVIATA')
  messaInMoraInviata,
  @JsonValue('CONTESTAZIONE_DA_RISCONTRARE')
  contestazioneDaRiscontrare,
  @JsonValue('DEPOSITO_RICORSO')
  depositoRicorso,
  @JsonValue('DECRETO_INGIUNTIVO_DA_NOTIFICARE')
  decretoIngiuntivoDaNotificare,
  @JsonValue('DECRETO_INGIUNTIVO_NOTIFICATO')
  decretoIngiuntivoNotificato,
  @JsonValue('PRECETTO')
  precetto,
  @JsonValue('PIGNORAMENTO')
  pignoramento,
  @JsonValue('COMPLETATA')
  completata,
}

extension CaseStateLabel on CaseState {
  String get label {
    switch (this) {
      case CaseState.messaInMoraDaFare:
        return 'Messa in Mora da Fare';
      case CaseState.messaInMoraInviata:
        return 'Messa in Mora Inviata';
      case CaseState.contestazioneDaRiscontrare:
        return 'Contestazione da Riscontrare';
      case CaseState.depositoRicorso:
        return 'Deposito Ricorso';
      case CaseState.decretoIngiuntivoDaNotificare:
        return 'DI da Notificare';
      case CaseState.decretoIngiuntivoNotificato:
        return 'DI Notificato';
      case CaseState.precetto:
        return 'Precetto';
      case CaseState.pignoramento:
        return 'Pignoramento';
      case CaseState.completata:
        return 'Completata';
    }
  }
}
